src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -1221,3 +1221,259 @@ class DMFT(ModelMethodElectronic):
+        """,
+        a_eln=ELNAnnotation(component='EnumEditQuantity'),
+    )
+
+    n_alpha_electrons = Quantity(
+        type=np.int32,
+        description="""
+        Number of alpha electrons (spin up) in the molecular system.
+        """,
+    )
+
+    n_beta_electrons = Quantity(
+        type=np.int32,
+        description="""
+        Number of beta electrons (spin down) in the molecular system.
+        """,
+    )
+
+    n_molecular_orbitals = Quantity(
+        type=np.int32,
+        description="""
+        Total number of molecular orbitals in the system.
+        """,
+    )
+
+    molecular_orbitals = SubSection(
+        sub_section=MolecularOrbital.m_def,
+        repeats=True,
+        description="""
+        Detailed information about each molecular orbital,
+        including energy, occupation, and symmetry label.
+        """,
+    )
+
+    total_spin = Quantity(
+        type=np.float64,
+        description="""
+        Total spin of the system defined as S = (n_alpha_electrons - n_beta_electrons) / 2.
+        Connect to the model system.
+        """,
+    )
+
+    spin_multiplicity = Quantity(
+        type=np.int32,
+        description="""
+        Spin multiplicity of the system defined as 2S + 1.
+        """,
+    )
+
+    def resolve_spin_properties(self, logger: 'BoundLogger') -> None:
+        """
+        Resolves the total spin and spin multiplicity of the system based on alpha and beta electrons.
+        """
+        if self.n_alpha_electrons is not None and self.n_beta_electrons is not None:
+            self.total_spin = (self.n_alpha_electrons - self.n_beta_electrons) / 2
+            self.spin_multiplicity = int(2 * self.total_spin + 1)
+
+    def validate_scheme(self, logger: 'BoundLogger') -> bool:
+        """
+        Validates the consistency of the molecular orbital scheme.
+
+        Returns:
+            (bool): True if the scheme is consistent, False otherwise.
+        """
+        if self.reference_type in ['RHF', 'RKS']:
+            if self.n_alpha_electrons != self.n_beta_electrons:
+                logger.error(
+                    f'For {self.reference_type}, the number of alpha and beta electrons must be equal.'
+                )
+                return False
+        if self.reference_type in ['ROHF', 'ROKS']:
+            if abs(self.n_alpha_electrons - self.n_beta_electrons) != 1:
+                logger.error(
+                    f'For {self.reference_type}, there must be exactly one unpaired electron.'
+                )
+                return False
+        return True
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+
+        # Resolve spin properties
+        self.resolve_spin_properties(logger)
+
+        # Validate the molecular orbital scheme
+        if not self.validate_scheme(logger):
+            logger.error('Invalid molecular orbital scheme.')
```

Comment:
Will just drop this link here: https://github.com/FAIRmat-NFDI/AreaC-Meetings/discussions/56
You can reply there about your take.

Reply:
@JFRudzinski you too, pls